src/server/JasmineGraphInstanceService.cpp

```diff
@@ -80,4301 +144,5243 @@ void *instanceservicesession(void *dummyPt) {
-            instance_logger.log("Received : In degree distribution to aggregator", "info");
+    if (JasmineGraphInstanceService::isInstanceCentralStoreExists(graphId, partitionId)) {
+        std::string centralStoreFile = dataFolder + "/" + centralGraphIdentifier;
+        std::string copyCommand;
 
-            write(connFd, JasmineGraphInstanceProtocol::OK.c_str(), JasmineGraphInstanceProtocol::OK.size());
-            instance_logger.log("Sent : " + JasmineGraphInstanceProtocol::OK, "info");
-            bzero(data, INSTANCE_DATA_LENGTH + 1);
-            read(connFd, data, INSTANCE_DATA_LENGTH);
-            string graphID = (data);
-            graphID = Utils::trim_copy(graphID, " \f\n\r\t\v");
-            instance_logger.log("Received Graph ID: " + graphID, "info");
+        DIR *dir = opendir(aggregatorFilePath.c_str());
+
+        if (dir) {
+            closedir(dir);
+        } else {
+            std::string createDirCommand = "mkdir -p " + aggregatorFilePath;
+            FILE *createDirInput = popen(createDirCommand.c_str(), "r");
+            pclose(createDirInput);
+        }
+
+        if (aggregatorHost == host) {
+            copyCommand = "cp " + centralStoreFile + " " + aggregatorFilePath;
+        } else {
+            copyCommand = "scp " + centralStoreFile + " " + aggregatorHost + ":" + aggregatorFilePath;
+        }
 
-            write(connFd, JasmineGraphInstanceProtocol::OK.c_str(), JasmineGraphInstanceProtocol::OK.size());
-            instance_logger.log("Sent : " + JasmineGraphInstanceProtocol::OK, "info");
+        FILE *copyInput = popen(copyCommand.c_str(), "r");
 
-            bzero(data, INSTANCE_DATA_LENGTH + 1);
-            read(connFd, data, INSTANCE_DATA_LENGTH);
-            string partitionID = (data);
-            partitionID = Utils::trim_copy(partitionID, " \f\n\r\t\v");
-            instance_logger.log("Received Partition ID: " + partitionID, "info");
+        if (copyInput) {
+            // read the input
+            while (!feof(copyInput)) {
+                if (fgets(buffer, 128, copyInput) != NULL) {
+                    result.append(buffer);
+                }
+            }
+            if (!result.empty()) {
+                std::cout << result << std::endl;
+            }
+            pclose(copyInput);
+        }
+    }
+
+    return result;
+}
+
+string JasmineGraphInstanceService::aggregateCentralStoreTriangles(std::string graphId, std::string partitionId,
+                                                                   std::string partitionIdList, int threadPriority) {
+    instance_logger.log("###INSTANCE### Started Aggregating Central Store Triangles", "info");
+    std::string aggregatorFilePath = Utils::getJasmineGraphProperty("org.jasminegraph.server.instance.aggregatefolder");
+    std::vector<std::string> fileNames;
+    map<long, unordered_set<long>> aggregatedCentralStore;
+    std::string centralGraphIdentifier = graphId + +"_centralstore_" + partitionId;
```

Comment:
```suggestion
    std::string centralGraphIdentifier = graphId + "_centralstore_" + partitionId;
```

Reply:
Fixed this and all similar occurrences of `+ +` in next PR

Reply:
Ok